packages/rulesets/src/spectral/test/patch-properties-correspond-to-put-properties.test.ts

```diff
@@ -0,0 +1,695 @@
+  return linter.run(myOpenApiDocument).then((results) => {
+    expect(results.length).toBe(1)
+    expect(results[0].message).toContain("Patch operations body cannot be empty.")
+    expect(results[0].path.join(".")).toBe("paths./foo.patch.parameters")
+  })
+})
+
+test("PatchPropertiesCorrespondToPutProperties should find errors if patch has properties and put body is not defined", () => {
+  const myOpenApiDocument = {
+    swagger: "2.0",
+    paths: {
+      "/foo": {
+        patch: {
+          tags: ["SampleTag"],
+          operationId: "Foo_Update",
+          description: "Test Description",
+          parameters: [
+            {
+              name: "foo_patch",
+              in: "body",
+              schema: {
+                $ref: "#/definitions/FooRequestParams",
+              },
+            },
+          ],
+          responses: {
+            "200": {
+              description: "success",
+              schema: {
+                $ref: "#/definitions/FooProps",
+              },
+            },
+          },
+        },
+        put: {
+          tags: ["SampleTag"],
+          operationId: "Foo_Update",
+          description: "Test Description",
+          parameters: [
+            {
+              name: "foo_put",
+              in: "path",
+            },
+          ],
+          responses: {
+            "200": {
+              description: "success",
+              schema: {
+                $ref: "#/definitions/FooResource",
+              },
+            },
+            "201": {
+              description: "newly created",
+              schema: {
+                $ref: "#/definitions/FooResource",
+              },
+            },
+          },
+        },
+      },
+    },
+    definitions: {
+      FooRequestParams: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooResource: {
+        "x-ms-azure-resource": true,
+      },
+      FooProps: {
+        properties: {
+          servicePrecedence: {
+            description:
+              "A precedence value that is used to decide between services when identifying the QoS values to use for a particular SIM. A lower value means a higher priority. This value should be unique among all services configured in the mobile network.",
+            type: "integer",
+            format: "int32",
+            minimum: 0,
+            maximum: 255,
+          },
+          id: {
+            type: "string",
+          },
+        },
+      },
+    },
+  }
+  return linter.run(myOpenApiDocument).then((results) => {
+    expect(results.length).toBe(1)
+    expect(results[0].message).toContain("Non empty patch body with an empty put body is not valid")
+    expect(results[0].path.join(".")).toBe("paths./foo.patch.parameters")
+  })
+})
+
+test("PatchPropertiesCorrespondToPutProperties should find errors when patch has additional property in deeply nested structure", () => {
+  const myOpenApiDocument = {
```

Comment:
BarRequestParams doe snot have deep nesting in the example you used. When you specify 'allOf' properties being referenced get inline at the same level